azurerm/internal/services/maintenance/tests/resource_arm_maintenance_configuration_test.go

```diff
@@ -0,0 +1,232 @@
+					resource.TestCheckResourceAttr(data.ResourceName, "scope", "All"),
+					resource.TestCheckResourceAttr(data.ResourceName, "tags.%", "0"),
+				),
+			},
+			{
+				Config: testAccAzureRMMaintenanceConfiguration_complete(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMMaintenanceConfigurationExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "scope", "Host"),
+					resource.TestCheckResourceAttr(data.ResourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(data.ResourceName, "tags.env", "test"),
+				),
+			},
+			{
+				Config: testAccAzureRMMaintenanceConfiguration_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMMaintenanceConfigurationExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "scope", "All"),
+					resource.TestCheckResourceAttr(data.ResourceName, "tags.%", "0"),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func testCheckAzureRMMaintenanceConfigurationDestroy(s *terraform.State) error {
+	conn := acceptance.AzureProvider.Meta().(*clients.Client).Maintenance.ConfigurationsClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_maintenance_configuration" {
+			continue
+		}
+
+		id, err := parse.MaintenanceConfigurationID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		resp, err := conn.Get(ctx, id.ResourceGroup, id.Name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return err
+			}
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testCheckAzureRMMaintenanceConfigurationExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := acceptance.AzureProvider.Meta().(*clients.Client).Maintenance.ConfigurationsClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		id, err := parse.MaintenanceConfigurationID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		resp, err := conn.Get(ctx, id.ResourceGroup, id.Name)
+		if err != nil {
+			return fmt.Errorf("Bad: Get on maintenanceConfigurationsClient: %+v", err)
+		}
+
+		if resp.StatusCode == http.StatusNotFound {
+			return fmt.Errorf("Bad: Maintenance Configuration %q (resource group: %q) does not exist", id.Name, id.ResourceGroup)
+		}
+
+		return nil
+	}
+}
+
+func testAccAzureRMMaintenanceConfiguration_basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_maintenance_configuration" "test" {
+  name                = "acctest-mc%d"
```

Comment:
could we ensure uppercase works
```suggestion
  name                = "acctest-MC%d"
```